packages/blitz/src/plugin.ts

```diff
@@ -0,0 +1,77 @@
+import {ClientPlugin} from "./index-browser"
+import {isClient} from "./utils"
+
+export function merge<T, U>([...fns]: Array<(args: T) => U>) {
+  return (args: T) => fns.map((fn) => fn(args))
+}
+
+export function pipe<T>(...fns: Array<(x: T) => T>) {
+  return (x: T) => fns.reduce((v, f) => f(v), x)
+}
+
+export function reduceBlitzPlugins<TPlugins extends readonly ClientPlugin<object>[]>({
```

Comment:
- [ ] Add `withProvider` to reduce
https://github.com/blitz-js/blitz/blob/8c247e26e7cba72d7d76dd83427a20604afc504f/packages/blitz-next/src/index-browser.tsx#L25-L29